common/lib/transport/connectionmanager.js

```diff
@@ -876,6 +871,15 @@ var ConnectionManager = (function() {
 		this.notifyState({state: 'closed'});
 	};
 
+	ConnectionManager.prototype.onAuthUpdated = function() {
+		var state = this.state.state;
+		if(state == 'connected' || state == 'connecting') {
+			/* in the current protocol version we are not able to update auth params on the fly;
+			 * so disconnect, and the new auth params will be used for subsequent reconnection */
+			this.disconnectAllTransports();
```

Comment:
I expect by doing this we'll trigger a disconnected state which in turn will:

* Trigger a disconnected event for users of this lib unexpectedly and they may as a result update their UI to reflect the disconnected state
* The client library will fallback to an HTTP Comet transport for it's first reconnect attempt.

I think both of these things are bad.